include/sir/ansimacros.h

```diff
@@ -68,68 +82,278 @@
  */
-# define COLOR(attr, fg, bg, s) \
-    SIR_ESC_SEQ(#attr ";" #fg ";" #bg, s) SIR_ESC_SEQE("0;39;49")
+# undef COLOR
+# if !defined(SIR_NO_TEXT_STYLING)
+#  define COLOR(attr, fg, bg, s) \
+     SIR_ESC_SEQ(#attr ";" #fg ";" #bg, s) SIR_ESC_SEQE("0;39;49")
+# else
+#  define COLOR(attr, fg, bg, s) s
+# endif
 
 # undef STRIKE
-# define STRIKE(s) SIR_ESC_SEQ("9", s) SIR_ESC_SEQE("29") /**< Strike-through. */
+# if !defined(SIR_NO_TEXT_STYLING)
+#  define STRIKE(s) SIR_ESC_SEQ("9", s) SIR_ESC_SEQE("29") /**< Strike-through. */
+# else
+#  define STRIKE(s) s
+# endif
+
 # undef INVERT
-# define INVERT(s) SIR_ESC_SEQ("7", s) SIR_ESC_SEQE("27") /**< Inverted fg/bg. */
+# if !defined(SIR_NO_TEXT_STYLING)
+#  define INVERT(s) SIR_ESC_SEQ("7", s) SIR_ESC_SEQE("27") /**< Inverted fg/bg. */
+# else
+#  define INVERT(s) s
+# endif
+
 # undef ULINE
-# define ULINE(s)  SIR_ESC_SEQ("4", s) SIR_ESC_SEQE("24") /**< Underlined. */
+# if !defined(SIR_NO_TEXT_STYLING)
+#  define ULINE(s) SIR_ESC_SEQ("4", s) SIR_ESC_SEQE("24") /**< Underlined. */
+# else
+#  define ULINE(s) s
+# endif
+
 # undef EMPH
-# define EMPH(s)   SIR_ESC_SEQ("3", s) SIR_ESC_SEQE("23") /**< Emphasis/italic. */
+# if !defined(SIR_NO_TEXT_STYLING)
+#  define EMPH(s) SIR_ESC_SEQ("3", s) SIR_ESC_SEQE("23") /**< Emphasis/italic. */
+# else
+#  define EMPH(s) s
+# endif
+
 # undef BOLD
-# define BOLD(s)   SIR_ESC_SEQ("1", s) SIR_ESC_SEQE("22") /**< Bold. */
+# if !defined(SIR_NO_TEXT_STYLING)
+#  define BOLD(s) SIR_ESC_SEQ("1", s) SIR_ESC_SEQE("22") /**< Bold. */
+# else
+#  define BOLD(s) s
+# endif
+
 # undef BLINK
-# define BLINK(s)  SIR_ESC_SEQ("5", s) SIR_ESC_SEQE("25") /**< Blinking text. */
+# if !defined(SIR_NO_TEXT_STYLING)
+#  define BLINK(s) SIR_ESC_SEQ("5", s) SIR_ESC_SEQE("25") /**< Blinking text. */
+# else
+#  define BLINK(s) s
+# endif
+
+# undef BLACK
+# if !defined(SIR_NO_TEXT_STYLING)
+#  define BLACK(s) COLOR(0, 30, 49, s) /**< Black foreground text. */
+# else
+#  define BLACK(s) s
+# endif
+
+# undef BLACKB
+# if !defined(SIR_NO_TEXT_STYLING)
+#  define BLACKB(s) COLOR(1, 30, 49, s) /**< Bold black foreground text. */
+# else
+#  define BLACKB(s) s
+# endif
+
+# undef RED
+# if !defined(SIR_NO_TEXT_STYLING)
+#  define RED(s) COLOR(0, 31, 49, s) /**< Red foreground text. */
+# else
+#  define RED(s) s
+# endif
+
+# undef REDB
+# if !defined(SIR_NO_TEXT_STYLING)
+#  define REDB(s) COLOR(1, 31, 49, s) /**< Bold red foreground text. */
+# else
+#  define REDB(s) s
+# endif
+
+# undef BRED
+# if !defined(SIR_NO_TEXT_STYLING)
+#  define BRED(s) COLOR(0, 91, 49, s) /**< Bright red foreground text. */
+# else
+#  define BRED(s) s
+# endif
+
+# undef BREDB
+# if !defined(SIR_NO_TEXT_STYLING)
+#  define BREDB(s) COLOR(1, 91, 49, s) /**< Bold bright red foreground text. */
+# else
+#  define BREDB(s) s
+# endif
+
+# undef GREEN
+# if !defined(SIR_NO_TEXT_STYLING)
+#  define GREEN(s) COLOR(0, 32, 49, s) /**< Green foreground text. */
+# else
+#  define GREEN(s) s
```

Comment:
You're going to hate me but I would like this better as a huge:

```c
#if !defined(SIR_NO_TEXT_STYLING)
 #define MACRO1(..., s) ...
 #define MACRO2(..., s) ...
 ...
#else
 #define MACRO1(..., s) s
 #define MACRO2(..., s) s
 ...
#endif
```
Rather than a bunch of #ifdefs

Reply:
*grumble grumble*

let me look at it.

Reply:
OK done.